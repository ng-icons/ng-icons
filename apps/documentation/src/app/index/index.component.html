<div
  appFadeInContainer
  class="flex flex-col items-center justify-center h-screen px-4 mt-6 md:mt-0"
>
  <img
    appFadeIn
    class="w-32 mb-6"
    src="assets/angular.svg"
    alt="Angular Logo"
  />

  <h2
    appFadeIn
    class="heading mb-5 text-base md:text-xl font-extrabold uppercase"
  >
    Angular Icons
  </h2>

  <h1
    appFadeIn
    class="slogan font-extrabold text-3xl md:text-5xl text-center mb-6 md:leading-[60px]"
  >
    The all-in-one icon<br />
    library for Angular
  </h1>

  <p appFadeIn class="text-sm md:text-base text-[#F5EAEA] text-center mb-16">
    Providing over 48,000 icons for use in your applications
  </p>

  <div appFadeIn class="flex gap-6 flex-col md:flex-row">
    <button
      routerLink="/getting-started"
      class="btn btn-white h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Get Started</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>

    <button
      routerLink="/browse-icons"
      class="btn btn-red h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Browse Icons</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>
  </div>
</div>
