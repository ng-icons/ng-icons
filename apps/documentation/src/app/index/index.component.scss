.heading {
  line-height: 25px;
  letter-spacing: 0.08em;
  background: linear-gradient(360deg, #ffafbd 0%, #ffc3a0 82%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.slogan {
  line-height: 60px;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    rgba(227, 153, 153, 0.96) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.btn {
  border-radius: 32px;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0);

  &.btn-white {
    background: linear-gradient(90deg, #e0eafc 0%, #cfdef3 100%);
    border-top: 1px solid #ffffff;
    color: #1a1e2a;

    &:hover {
      box-shadow: 0px 1px 40px rgba(224, 234, 252, 0.4);
    }
  }

  &.btn-red {
    border-top: 1px solid rgba(238, 238, 238, 0.3);
    background: linear-gradient(90deg, #e53935 0%, #e35d5b 100%);
    color: #fff;

    &:hover {
      box-shadow: 0px 1px 40px rgba(216, 24, 47, 0.5);
    }
  }
}
