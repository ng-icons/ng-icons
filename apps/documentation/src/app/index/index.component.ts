import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NgIconComponent, provideIcons } from '@ng-icons/core';
import { heroChevronRightMini } from '@ng-icons/heroicons/mini';
import { FadeInContainerDirective } from '../directives/fade-in/fade-in-container.directive';
import { FadeInDirective } from '../directives/fade-in/fade-in.directive';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss'],
  standalone: true,
  imports: [
    NgIconComponent,
    RouterLink,
    FadeInContainerDirective,
    FadeInDirective,
  ],
  providers: [provideIcons({ heroChevronRightMini })],
})
export class IndexComponent {}
