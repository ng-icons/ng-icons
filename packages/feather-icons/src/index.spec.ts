import * as icons from './index';

describe('Feather Icons', () => {
  it('should ensure the export names have not changed unexpectedly', () => {
    expect(Object.keys(icons)).toMatchSnapshot();
  })
});
