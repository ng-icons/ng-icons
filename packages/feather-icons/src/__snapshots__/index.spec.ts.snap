// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feather Icons should ensure the export names have not changed unexpectedly 1`] = `
Array [
  "featherActivity",
  "featherAirplay",
  "featherAlertCircle",
  "featherAlertOctagon",
  "featherAlertTriangle",
  "featherAlignCenter",
  "featherAlignJustify",
  "featherAlignLeft",
  "featherAlignRight",
  "featherAnchor",
  "featherAperture",
  "featherArchive",
  "featherArrowDownCircle",
  "featherArrowDownLeft",
  "featherArrowDownRight",
  "featherArrowDown",
  "featherArrowLeftCircle",
  "featherArrowLeft",
  "featherArrowRightCircle",
  "featherArrowRight",
  "featherArrowUpCircle",
  "featherArrowUpLeft",
  "featherArrowUpRight",
  "featherArrowUp",
  "featherAtSign",
  "featherAward",
  "featherBarChart2",
  "featherBarChart",
  "featherBatteryCharging",
  "featherBattery",
  "featherBellOff",
  "featherBell",
  "featherBluetooth",
  "featherBold",
  "featherBookOpen",
  "featherBook",
  "featherBookmark",
  "featherBox",
  "featherBriefcase",
  "featherCalendar",
  "featherCameraOff",
  "featherCamera",
  "featherCast",
  "featherCheckCircle",
  "featherCheckSquare",
  "featherCheck",
  "featherChevronDown",
  "featherChevronLeft",
  "featherChevronRight",
  "featherChevronUp",
  "featherChevronsDown",
  "featherChevronsLeft",
  "featherChevronsRight",
  "featherChevronsUp",
  "featherChrome",
  "featherCircle",
  "featherClipboard",
  "featherClock",
  "featherCloudDrizzle",
  "featherCloudLightning",
  "featherCloudOff",
  "featherCloudRain",
  "featherCloudSnow",
  "featherCloud",
  "featherCode",
  "featherCodepen",
  "featherCodesandbox",
  "featherCoffee",
  "featherColumns",
  "featherCommand",
  "featherCompass",
  "featherCopy",
  "featherCornerDownLeft",
  "featherCornerDownRight",
  "featherCornerLeftDown",
  "featherCornerLeftUp",
  "featherCornerRightDown",
  "featherCornerRightUp",
  "featherCornerUpLeft",
  "featherCornerUpRight",
  "featherCpu",
  "featherCreditCard",
  "featherCrop",
  "featherCrosshair",
  "featherDatabase",
  "featherDelete",
  "featherDisc",
  "featherDivideCircle",
  "featherDivideSquare",
  "featherDivide",
  "featherDollarSign",
  "featherDownloadCloud",
  "featherDownload",
  "featherDribbble",
  "featherDroplet",
  "featherEdit2",
  "featherEdit3",
  "featherEdit",
  "featherExternalLink",
  "featherEyeOff",
  "featherEye",
  "featherFacebook",
  "featherFastForward",
  "featherFeather",
  "featherFigma",
  "featherFileMinus",
  "featherFilePlus",
  "featherFileText",
  "featherFile",
  "featherFilm",
  "featherFilter",
  "featherFlag",
  "featherFolderMinus",
  "featherFolderPlus",
  "featherFolder",
  "featherFramer",
  "featherFrown",
  "featherGift",
  "featherGitBranch",
  "featherGitCommit",
  "featherGitMerge",
  "featherGitPullRequest",
  "featherGithub",
  "featherGitlab",
  "featherGlobe",
  "featherGrid",
  "featherHardDrive",
  "featherHash",
  "featherHeadphones",
  "featherHeart",
  "featherHelpCircle",
  "featherHexagon",
  "featherHome",
  "featherImage",
  "featherInbox",
  "featherInfo",
  "featherInstagram",
  "featherItalic",
  "featherKey",
  "featherLayers",
  "featherLayout",
  "featherLifeBuoy",
  "featherLink2",
  "featherLink",
  "featherLinkedin",
  "featherList",
  "featherLoader",
  "featherLock",
  "featherLogIn",
  "featherLogOut",
  "featherMail",
  "featherMapPin",
  "featherMap",
  "featherMaximize2",
  "featherMaximize",
  "featherMeh",
  "featherMenu",
  "featherMessageCircle",
  "featherMessageSquare",
  "featherMicOff",
  "featherMic",
  "featherMinimize2",
  "featherMinimize",
  "featherMinusCircle",
  "featherMinusSquare",
  "featherMinus",
  "featherMonitor",
  "featherMoon",
  "featherMoreHorizontal",
  "featherMoreVertical",
  "featherMousePointer",
  "featherMove",
  "featherMusic",
  "featherNavigation2",
  "featherNavigation",
  "featherOctagon",
  "featherPackage",
  "featherPaperclip",
  "featherPauseCircle",
  "featherPause",
  "featherPenTool",
  "featherPercent",
  "featherPhoneCall",
  "featherPhoneForwarded",
  "featherPhoneIncoming",
  "featherPhoneMissed",
  "featherPhoneOff",
  "featherPhoneOutgoing",
  "featherPhone",
  "featherPieChart",
  "featherPlayCircle",
  "featherPlay",
  "featherPlusCircle",
  "featherPlusSquare",
  "featherPlus",
  "featherPocket",
  "featherPower",
  "featherPrinter",
  "featherRadio",
  "featherRefreshCcw",
  "featherRefreshCw",
  "featherRepeat",
  "featherRewind",
  "featherRotateCcw",
  "featherRotateCw",
  "featherRss",
  "featherSave",
  "featherScissors",
  "featherSearch",
  "featherSend",
  "featherServer",
  "featherSettings",
  "featherShare2",
  "featherShare",
  "featherShieldOff",
  "featherShield",
  "featherShoppingBag",
  "featherShoppingCart",
  "featherShuffle",
  "featherSidebar",
  "featherSkipBack",
  "featherSkipForward",
  "featherSlack",
  "featherSlash",
  "featherSliders",
  "featherSmartphone",
  "featherSmile",
  "featherSpeaker",
  "featherSquare",
  "featherStar",
  "featherStopCircle",
  "featherSun",
  "featherSunrise",
  "featherSunset",
  "featherTable",
  "featherTablet",
  "featherTag",
  "featherTarget",
  "featherTerminal",
  "featherThermometer",
  "featherThumbsDown",
  "featherThumbsUp",
  "featherToggleLeft",
  "featherToggleRight",
  "featherTool",
  "featherTrash2",
  "featherTrash",
  "featherTrello",
  "featherTrendingDown",
  "featherTrendingUp",
  "featherTriangle",
  "featherTruck",
  "featherTv",
  "featherTwitch",
  "featherTwitter",
  "featherType",
  "featherUmbrella",
  "featherUnderline",
  "featherUnlock",
  "featherUploadCloud",
  "featherUpload",
  "featherUserCheck",
  "featherUserMinus",
  "featherUserPlus",
  "featherUserX",
  "featherUser",
  "featherUsers",
  "featherVideoOff",
  "featherVideo",
  "featherVoicemail",
  "featherVolume1",
  "featherVolume2",
  "featherVolumeX",
  "featherVolume",
  "featherWatch",
  "featherWifiOff",
  "featherWifi",
  "featherWind",
  "featherXCircle",
  "featherXOctagon",
  "featherXSquare",
  "featherX",
  "featherYoutube",
  "featherZapOff",
  "featherZap",
  "featherZoomIn",
  "featherZoomOut",
]
`;
